src/components/ui/ImageGallery/ImageGallerySelector.tsx

```diff
@@ -0,0 +1,130 @@
+import type { HTMLAttributes, PropsWithChildren } from 'react'
+import React, { useMemo } from 'react'
+import { useSlider, IconButton } from '@faststore/ui'
+
+import { ForwardArrowIcon, BackwardArrowIcon } from './Icons'
+
+interface Props extends HTMLAttributes<HTMLDivElement> {
+  itemsPerPage: number
+}
+
+const createTransformValues = (totalItems: number) => {
+  if (!totalItems) {
+    return {}
+  }
+
+  const transformMap: Record<number, number> = {}
+  const slideWidth = 100 / totalItems
+
+  for (let idx = 0; idx < totalItems; ++idx) {
+    const transformValue = -(slideWidth * idx)
+
+    transformMap[idx] = transformValue
+  }
+
+  return transformMap
+}
+
+function ImageGallerySelector({
+  itemsPerPage,
+  children,
+  ...otherProps
+}: PropsWithChildren<Props>) {
+  const elements = React.Children.toArray(children)
+  const elementCount = elements.length
+
+  const transformValues = useMemo(
+    () => createTransformValues(elementCount),
+    [elementCount]
+  )
+
+  const { handlers, slide, sliderState, sliderDispatch } = useSlider({
+    totalItems: elementCount,
+    itemsPerPage,
+    infiniteMode: false,
+  })
+
+  const slidingTransition = `transform 400ms`
+
+  if (!elements || !elementCount) {
```

Comment:
```suggestion
  if (!elementsCount) {
```
I think the first condition (![]) is always false since the array is an object and is always evaluated as a [truthy](https://developer.mozilla.org/pt-BR/docs/Glossary/Truthy) value, but with the `!` operator will be always `falsy`. 

I also wonder if we should move this guard clause to the beginning of the function.

Reply:
If we try to move the guard clause to the beginning of the function, this error is reported from the linter:

> _React Hook "useSlider" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?_
(similar error arises for `useMemo`)

https://reactjs.org/docs/hooks-rules.html

Reply:
About that:

> I think the first condition (![]) is always false since the array is an object and is always evaluated as a [truthy](https://developer.mozilla.org/pt-BR/docs/Glossary/Truthy) value, but with the `!` operator will be always `falsy`.

You are right. I thought that if the children happened to be `[]`, `null` or `undefined`, the `.toArray` method would behave unexpectedly, but it always returns an empty array. And like you said, `![]` is always `false`. Fixed at